iree/compiler/Dialect/HAL/Target/TargetBackend.cpp

```diff
@@ -84,6 +84,29 @@ BufferConstraintsAttr TargetBackend::queryBufferConstraints(
   return makeDefaultBufferConstraints(context);
 }
 
+// Renames |op| within |moduleOp| with a new name that is unique within both
+// |moduleOp| and |otherModuleOp|.
+static void renameWithDisambiguatedName(Operation *op, Operation *moduleOp,
+                                        Operation *otherModuleOp,
+                                        llvm::StringRef originalName) {
+  SymbolTable moduleSymbolTable(moduleOp);
```

Comment:
this is really expensive to create (it walks all the ops in the module to build the table each time) - you'll need to do this once and reuse it - you should only build one symbol table per module

Reply:
you may not even need either of them - you have your targetSymbolMap - just check that - if it exists try again - no need for SymbolTables at all

Reply:
sorry I didn't spot this on discord!

Reply:
Nice, builds all passed. So this code works as expected, just need to polish it a bit :)

Reply:
I tried to make some deeper changes to how symbols are tracked, just calling `SymbolTable::insert()`, but figured that custom logic would still be needed to handle the case where a public symbol is inserted on top of an already added private symbol and the private symbol needs to be renamed to avoid conflicts within the target module and to avoid conflicts with the public symbol to be inserted.

Short of that, I settled for reusing a `SymbolTable` and `targetSymbolMap` per your suggestions. Re-running tests now (https://buildkite.com/iree/iree-android-benchmark/builds/2683) 🤞 

I also reverted https://github.com/google/iree/pull/6067 (to re-enable some tests that should pass again with this roll-forward)

Reply:
Whoops. That broke the real model tests again :P

Standby >_>

Reply:
Fixed. PTAL.